src/main/java/com/spotify/sparkey/Util.java

```diff
@@ -270,4 +270,32 @@ static void nonThrowingClose(Closeable stream) {
       e.printStackTrace();
     }
   }
+
+  static boolean renameFile(File srcFile, File destFile) {
+    if (!destFile.exists()) {
+      return srcFile.renameTo(destFile);
+    }
+
+    File backupFile = new File(destFile.getParent(), destFile.getName() + "-backup" + System.currentTimeMillis());
+    if (backupFile.exists()) {
+      // Edge case, but let's be safe.
+      return false;
```

Comment:
Minor Comment: Before giving up, may be we should try few more times  and/or ensure filenames have *more* randomness/ uniqueness. I understand that probability is less though.